inlong-agent/agent-common/src/main/java/org/apache/inlong/agent/utils/EventReportUtils.java

```diff
@@ -0,0 +1,108 @@
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.apache.inlong.agent.constant.AgentConstants.AGENT_LOCAL_IP;
+import static org.apache.inlong.agent.metrics.AgentEventMetricItem.KEY_INLONG_AGENT_IP;
+import static org.apache.inlong.agent.metrics.AgentEventMetricItem.KEY_INLONG_COMPONENT_NAME;
+import static org.apache.inlong.agent.metrics.AgentEventMetricItem.KEY_INLONG_COMPONENT_TYPE;
+import static org.apache.inlong.agent.metrics.AgentEventMetricItem.KEY_INLONG_COMPONENT_VERSION;
+import static org.apache.inlong.agent.metrics.AgentEventMetricItem.KEY_INLONG_EVENT_CODE;
+import static org.apache.inlong.agent.metrics.AgentEventMetricItem.KEY_INLONG_EVENT_DESC;
+import static org.apache.inlong.agent.metrics.AgentEventMetricItem.KEY_INLONG_EVENT_LEVEL;
+import static org.apache.inlong.agent.metrics.AgentEventMetricItem.KEY_INLONG_EVENT_TIME;
+import static org.apache.inlong.agent.metrics.AgentEventMetricItem.KEY_INLONG_EVENT_TYPE;
+import static org.apache.inlong.agent.metrics.AgentEventMetricItem.KEY_INLONG_EXT;
+import static org.apache.inlong.agent.metrics.AgentEventMetricItem.KEY_INLONG_GROUP_ID;
+import static org.apache.inlong.agent.metrics.AgentEventMetricItem.KEY_INLONG_STREAM_ID;
+
+/**
+ * DiagUtils
+ */
+public class EventReportUtils {
+
+    public enum EvenCodeEnum {
+
+        CONFIG_UPDATE_SUC(0, "config update suc"),
+        CONFIG_NO_UPDATE(1, "config no update"),
+        CONFIG_UPDATE_VERSION_NO_CHANGE(2, "config update version no change"),
+        CONFIG_INVALID_RET_CODE(3, "config invalid ret code"),
+        CONFIG_INVALID_RESULT(4, "config invalid result maybe visit manager failed"),
+        TASK_ADD(5, "task add"),
+        TASK_DELETE(6, "task delete");
+
+        private final int code;
+        private final String message;
+
+        EvenCodeEnum(int code, String message) {
+            this.code = code;
+            this.message = message;
+        }
+
+        public int getCode() {
+            return code;
+        }
+
+        public String getMessage() {
+            return message;
+        }
+    }
+
+    private final static String COMPONENT_TYPE_AGENT = "AGENT";
+    private final static String COMPONENT_NAME_AGENT = "AGENT";
+    public static final String EVENT_TYPE_CONFIG_UPDATE = "CONFIG_UPDATE";
+    public static final String EVENT_LEVEL_INFO = "INFO";
+    public static final String EVENT_LEVEL_WARN = "WARN";
+    public static final String EVENT_LEVEL_ERROR = "ERROR";
+    private static AgentEventMetricItemSet metricItemSet;
+
+    private EventReportUtils() {
+    }
+
+    public static void init() {
+        metricItemSet = new AgentEventMetricItemSet(COMPONENT_NAME_AGENT);
+        MetricRegister.register(metricItemSet);
+    }
+
+    public static void report(String groupId, String streamId, long eventTime, String eventType,
+            String eventLevel, EvenCodeEnum evenCode, String ext, String desc) {
+        Map<String, String> dims = new HashMap<>();
```

Comment:
It the most that groupId and streamId are empty.
It is better to add new method.
public static void reportError(long eventTime, String eventType, String eventLevel, EvenCodeEnum evenCode, String ext, String desc)

Reply:
There will be many groupid streams later that are not empty